test_custom_querier.sh

```diff
@@ -21,28 +21,62 @@ echo $RES
+RES=$(${BIN} tx wasm execute $ICQ_CONTRACT_ADDRESS '{"register_government_proposals_query":{"connection_id":"connection-0", "proposals_ids": [1, 2],"update_period":10}}' --from ${KEY}  -y --chain-id ${CHAINID} --output json --broadcast-mode=block --gas-prices 0.0025stake --gas 1000000 --keyring-backend test --home ${HOME} --node tcp://127.0.0.1:16657)
+echo $RES | jq
 
 ${BIN} tx bank send demowallet2 neutron17dtl0mjt3t77kpuhg2edqzjpszulwhgzcdvagh 10000stake --chain-id ${CHAINID2} --home ${HOME2} --node tcp://localhost:26657 --keyring-backend test -y --gas-prices 0.0025stake --broadcast-mode=block
 
 ${BIN} tx staking delegate neutronvaloper1qnk2n4nlkpw9xfqntladh74w6ujtulwnqshepx 1000stake --chain-id ${CHAINID2} --home ${HOME2} --node tcp://localhost:26657 --keyring-backend test -y --gas-prices 0.0025stake --broadcast-mode=block  --from demowallet2
 
-
+echo "${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{\"get_registered_query\":{\"query_id\":1}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657"
 #note: query id is hardcodded into the query
 RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"get_registered_query":{"query_id":1}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
-echo $RES
+echo $RES | jq
 
-sleep 15
+sleep 30
 
 RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"balance": {"query_id": 1}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
-echo $RES
+echo $RES | jq
 
 #demowallet2 - neutron10h9stc5v6ntgeygf5xf945njqq5h32r54rf7kf
 RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"get_delegations":{"query_id": 2}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
-echo $RES
+echo $RES | jq
+
+RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"bank_total_supply":{"query_id": 3}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
+echo $RES | jq
+
+RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"staking_validators":{"query_id": 4}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
+echo $RES | jq
+
+RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"distribution_fee_pool":{"query_id": 5}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
+echo $RES | jq
+
+RES=$(${BIN} q wasm contract-state smart $ICQ_CONTRACT_ADDRESS '{"government_proposals":{"query_id": 6}}' --chain-id ${CHAINID} --output json  --home ${HOME} --node tcp://127.0.0.1:16657)
+echo $RES | jq
```

Comment:
Every single one of these queries fail for me: `Generic error: Querier contract error: codespace: interchainqueries, code: 1115: query wasm contract failed: invalid request`.

Stack trace:
```
Error: rpc error: code = InvalidArgument desc = 
github.com/cosmos/cosmos-sdk/baseapp.gRPCErrorToSDKError
        github.com/cosmos/cosmos-sdk@v0.45.5/baseapp/abci.go:590
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).handleQueryGRPC
        github.com/cosmos/cosmos-sdk@v0.45.5/baseapp/abci.go:579
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Query
        github.com/cosmos/cosmos-sdk@v0.45.5/baseapp/abci.go:421
github.com/tendermint/tendermint/abci/client.(*localClient).QuerySync
        github.com/tendermint/tendermint@v0.34.19/abci/client/local_client.go:256
github.com/tendermint/tendermint/proxy.(*appConnQuery).QuerySync
        github.com/tendermint/tendermint@v0.34.19/proxy/app_conn.go:159
github.com/tendermint/tendermint/rpc/core.ABCIQuery
        github.com/tendermint/tendermint@v0.34.19/rpc/core/abci.go:20
reflect.Value.call
        reflect/value.go:556
reflect.Value.Call
        reflect/value.go:339
github.com/tendermint/tendermint/rpc/jsonrpc/server.makeJSONRPCHandler.func1
        github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_json_handler.go:96
github.com/tendermint/tendermint/rpc/jsonrpc/server.handleInvalidJSONRPCPaths.func1
        github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_json_handler.go:122
net/http.HandlerFunc.ServeHTTP
        net/http/server.go:2084
net/http.(*ServeMux).ServeHTTP
        net/http/server.go:2462
github.com/tendermint/tendermint/rpc/jsonrpc/server.maxBytesHandler.ServeHTTP
        github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_server.go:236
github.com/tendermint/tendermint/rpc/jsonrpc/server.RecoverAndLogHandler.func1
        github.com/tendermint/tendermint@v0.34.19/rpc/jsonrpc/server/http_server.go:209
net/http.HandlerFunc.ServeHTTP
        net/http/server.go:2084
net/http.serverHandler.ServeHTTP
        net/http/server.go:2916
net/http.(*conn).serve
        net/http/server.go:1966
```

Reply:
you got this error because `neutron-query-relayer` and this script was using same address for operations. So if you will take a look a little upper you will see that some of previous requests was failed due to sequence mismatch. But I fixed this issue.

Reply:
I wrote about it in my previous comment https://github.com/neutron-org/neutron-contracts/pull/35#discussion_r1001652931

Reply:
It is reproducible 100% of a time for me. Or maybe I have been extremely unlucky?

Reply:
That's because separate address was not added to the script. Try it now

Reply:
It was discussed here https://github.com/neutron-org/neutron-contracts/pull/35#pullrequestreview-1147341564 but was not implement until now.

Reply:
> That's because separate address was not added to the script. Try it now

Honestly I thought the script has been working half of the time, not _always_ failing. I will try running it right now, thank you!

Reply:
Still failing for me! I run
```bash
neutrond query interchainqueries registered-queries --node tcp://127.0.0.1:16657 --output json | jq '.registered_queries[] | {id,last_submitted_result_local_height,last_submitted_result_remote_height}'
````
to check out if there are query results for all registered queries, but there are not all of them:
```json
{
  "id": "1",
  "last_submitted_result_local_height": "261",
  "last_submitted_result_remote_height": "258"
}
{
  "id": "2",
  "last_submitted_result_local_height": "0",
  "last_submitted_result_remote_height": "0"
}
{
  "id": "3",
  "last_submitted_result_local_height": "0",
  "last_submitted_result_remote_height": "0"
}
{
  "id": "4",
  "last_submitted_result_local_height": "0",
  "last_submitted_result_remote_height": "0"
}
{
  "id": "5",
  "last_submitted_result_local_height": "255",
  "last_submitted_result_remote_height": "252"
}
{
  "id": "6",
  "last_submitted_result_local_height": "258",
  "last_submitted_result_remote_height": "255"
}
```

Reply:
> Still failing for me! I run
> ....

Hmm.. it works for me.
```
{
  "id": "1",
  "last_submitted_result_local_height": "147",
  "last_submitted_result_remote_height": "145"
}
{
  "id": "2",
  "last_submitted_result_local_height": "138",
  "last_submitted_result_remote_height": "136"
}
{
  "id": "3",
  "last_submitted_result_local_height": "138",
  "last_submitted_result_remote_height": "136"
}
{
  "id": "4",
  "last_submitted_result_local_height": "140",
  "last_submitted_result_remote_height": "138"
}
{
  "id": "5",
  "last_submitted_result_local_height": "140",
  "last_submitted_result_remote_height": "138"
}
{
  "id": "6",
  "last_submitted_result_local_height": "144",
  "last_submitted_result_remote_height": "142"
}
```

can you give me your sequence of actions please?

Reply:
1. clone neutron with audit fixes branch and run `make init && make start-rly`
2. clone neutron-query-relayer with main branch and run `export $(grep -v '^#' .env.example.dev | xargs) && make dev`
3. go to neutron-contracts with your branch and run `./test_custom_querier.sh`

Reply:
gosh, I had to clean relayer's storage… everything works now.